File: benchmark-bus.cc
#include <benchmark/benchmark.h>
#include "my_message_bus.h"

void sendAndReceiveMessage(MessageBus& bus) {
    capnp::MessageBuilder message;
    // ... build the message ...
    bus.sendMessage(message);

    auto response = bus.receiveMessage();
    // ... process the response ...
}

BENCHMARK(sendAndReceiveMessage);

File: benchmark-bus.py
import pytest
from my_message_bus import MessageBus

@pytest.mark.benchmark
def test_message_bus_latency(benchmark):
    bus = MessageBus()
    bus.connect("tcp://localhost:5555")

    def send_and_receive_message():
        bus.send_message({"key": "value"})
        response = bus.receive_message()
        return response

    benchmark(send_and_receive_message)
